### ISSUE TITLE
Refactor web server shutdown procedure

### ISSUE THREAD

**Issue by asvetlov** [aio-libs/aiohttp#384]
Now we should use complicated code for graceful shutdown:

```
loop = asyncio.get_event_loop()
handler = app.make_handler()
f = loop.create_server(handler, '0.0.0.0', 8080)
srv = loop.run_until_complete(f)
print('serving on', srv.sockets[0].getsockname())
try:
    loop.run_forever()
except KeyboardInterrupt:
    pass
finally:
    loop.run_until_complete(handler.finish_connections(1.0))
    srv.close()
    loop.run_until_complete(srv.wait_closed())
    loop.run_until_complete(app.finish())
loop.close()
```

The problem is:
1. `finally` block is very complex
2. It doesn't prevent establishing new connections by client on `handler.finish_connections()` call.

I guess add new API like:

```
finalizer = yield from app.create_server(...)
try:
    loop.run_forever()
finally:
    yield from finalizer 
```

**Comment by ludovic-gasc** [comment]
+2

**Comment by asvetlov** [comment]
Well, finalizer api should be more complex:

```
finalizer.close(3.0)
yield from finalizer.wait_closed()
```

**Comment by asvetlov** [comment]
Or even

```
app.close(3.0)
yield from app.wait_closed()
```

to mimic [asyncio server protocol](https://docs.python.org/3/library/asyncio-eventloop.html#server) and to avoid inventing new `Finalizer` class.

**Comment by fafhrd91** [comment]
-10, "establishing new connections" is not a concern of application. you want to copy whole create_server api from asyncio?

**Comment by asvetlov** [comment]
Fixed by #734
